### PR TITLE
[dynamo] Support __contains__ on  __dict__ on UserDefinedClassVariable

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #131414
* #131405
* #131389
* __->__ #131378
* #131367
* #131347

Fixes https://github.com/pytorch/pytorch/issues/129665

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Review by mlazos — APPROVED** [review]
I'm fine with this as is, but couldn't we support retrieving `__dict__` on all variable types?

File: torch/_dynamo/variables/misc.py [hunk 1]
```diff
@@ -758,7 +758,11 @@ def call_method(
             and args[0].is_python_constant()
             and isinstance(
                 self.obj,
```
Comment (mlazos) [inline torch/_dynamo/variables/misc.py:760]: Why have this restriction?

Comment (anijain2305) [reply torch/_dynamo/variables/misc.py:760]: Good point! We should do that, we might need a method that we need to add to all variable trackers.

**Pull request closed without merging**